src/components/UI/Search.js

```diff
@@ -1,17 +1,56 @@
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
 import classNames from "classnames";
 import styles from "./Search.module.css";
+import {
+  faCircleXmark,
+  faMagnifyingGlass,
+} from "@fortawesome/free-solid-svg-icons";
+import { useState } from "react";
 
 const DEFAULT_DELAY = 1000;
 
 const Search = (props) => {
-  const changeHandler = (event) => {
+  const [value, setValue] = useState("");
+  const [isEmpty, setIsEmpty] = useState(true);
+
+  const changeHandler = ({ target: { value } }) => {
```

Comment:
May not work, `event.target` not always spcified. 

`.value` is set only for inputs, except CheckBoxes

Better code is

```
const changeHandler = useCallback((event) => {
   changeWithDebonce(event?.target?.value ?? '')
}, [changeWithDebonce])

```